### PR TITLE
New options: no file, archive and git deps

### PR DESCRIPTION
**Pull request by miripiruni**
**Motivation**

I want to be able check dependencies that unreliable.

Developers can use deps from git branches, tags or even commit sha. Such deps can be corrupted (after force push, repo deletion or move from user to organisation).

It‘s more reliable to use npm packages (since it’s hard to unpublish).

So, I introduced options:
— no-file-dependencies (from local directory)
— no-archive-dependencies (from tar or zip)
— no-git-dependencies (including commits sha, tags and branches)

The same options for devDeps.

**Checklist**

  - [x] Unit tests have been added
  - [x] Specific notes for documentation, if applicable

### CODE REVIEW

File: test/unit/validators/dependency-audit.test.js [hunk 1]
```diff
@@ -744,6 +744,120 @@ describe('dependency-audit Unit Tests', () => {
     });
   });
 
+  describe('doVersContainGitRepository method', () => {
```
Comment (miripiruni) [inline test/unit/validators/dependency-audit.test.js]: Question about `dependency-audit.test.js`: why we need this tests since it’s duplicate tests from `test/unit/rules/*`?

Now I carefully write code and tests at the same style as already written. But for me it’s a little bit overtested :)

**Comment by tclindner** [comment]
Thank you for opening a PR! I wanted to let you know that I won’t be able to look at this for the next week. I’m looking forward to reviewing.

**Comment by miripiruni** [comment]
@tclindner thank you for quick response. I’ll wait.

**Comment by miripiruni** [comment]
polite ping

**Comment by tclindner** [comment]
Hey @miripiruni I apologize for the delay! I was traveling ✈️ 😄 

This is fantastic! 🎉 These rules are awesome. You did a fantastic job on them. 🙌 I'll get a release cut shortly. 

Thank you again! I really appreciate all of your effort.

File: test/unit/validators/dependency-audit.test.js (hunk 1 above)
Comment (tclindner) [reply test/unit/validators/dependency-audit.test.js]: Hey @miripiruni - great question. You are right that it is probably a bit overkill. I figure it is better to be safe 🤓 . I'll think about adjustments for v5.
